### PR TITLE
Feature/12246 delta enf logic/12347 mark packages as checked for exposure

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Keep fingerprints, inject keyPackageDownloader and mark packages as processed in store
Note that the written files will rebuild completely if a package is not marked as processed, so we can delete all of them

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12347

### CODE REVIEW

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/__tests__/ExposureDetectionTests.swift [hunk 1]
```diff
@@ -17,7 +17,11 @@ final class ExposureDetectionTransactionTests: CWATestCase {
 		try "url0".write(to: url0, atomically: true, encoding: .utf8)
 		try "url1".write(to: url1, atomically: true, encoding: .utf8)
 
-		let writtenPackages = WrittenPackages(urls: [url0, url1])
+
+		let package01 = PackageContainer(hash: "1234", type: .signature, url: url0)
+		let package02 = PackageContainer(hash: "5678", type: .keys, url: url1)
+
+		let writtenPackages = WrittenPackages([package01, package02])
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/__tests__/ExposureDetectionTests.swift:24]: 👍

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift [hunk 2]
```diff
@@ -42,54 +69,30 @@ final class ExposureDetection {
...
-				self.detectExposureWindows(writtenPackages: writtenPackages, exposureConfiguration: exposureConfiguration)
+			writtenPackages: writtenPackages,
+			completion: { [weak self] result in
+				defer {
+					writtenPackages.cleanUp()
+				}
+
+				switch result {
+				case .success(let exposureWindows):
+					keyPackageDownload.markPackagesAsCheckedForExposures(writtenPackages.fingerprints)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:88]: I think its only needed to pass the fingerprints from the packages of type keys.

File: src/xcode/ENA/ENA/Source/Services/Risk/KeyPackageDownload/KeyPackageDownload.swift [hunk 3]
```diff
@@ -119,6 +121,14 @@ class KeyPackageDownload: KeyPackageDownloadProtocol {
 		}
 	}
 
+	func markPackagesAsCheckedForExposures(_ fingerprints: [String]) {
+		do {
+			try downloadedPackagesStore.markPackagesAsCheckedForExposures(fingerprints)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/Risk/KeyPackageDownload/KeyPackageDownload.swift:126]: Would it make sense to pass the store to the ExposureDetection? Then this func would not be needed.

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift (hunk 2 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:88]: I added a filter to the fingerprints property in WrittenPackages

File: src/xcode/ENA/ENA/Source/Services/Risk/KeyPackageDownload/KeyPackageDownload.swift (hunk 3 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/Risk/KeyPackageDownload/KeyPackageDownload.swift]: I made the requested changed

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift [hunk 4]
```diff
@@ -26,39 +44,57 @@ struct WrittenPackages {
 		}
 	}
 
-	mutating func add(_ url: URL) {
-		urls.append(url)
+	mutating func add(_ container: PackageContainer) {
+		packages.append(container)
+	}
+
+	var fingerprints: [String] {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift:51]:
```suggestion
	var keyFingerprints: [String] {
```

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift (hunk 4 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift]: 👍

**Pull request merged**